plugins/glustershd/glustershd.go

```diff
@@ -0,0 +1,102 @@
+package glustershd
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"os/exec"
+	"path"
+
+	"github.com/cespare/xxhash"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+
+	config "github.com/spf13/viper"
+)
+
+const (
+	glustershdBin = "glusterfs"
+)
+
+// Glustershd type represents information about Glustershd process
+type Glustershd struct {
+	binarypath     string
+	args           string
+	socketfilepath string
+	pidfilepath    string
+}
+
+// Name returns human-friendly name of the glustershd process. This is used for
+// logging
+func (b *Glustershd) Name() string {
+	return "glustershd"
+}
+
+// Path returns absolute path of the binary of glustershd process
+func (b *Glustershd) Path() string {
+	return b.binarypath
+}
+
+// Args returns arguments to be passed to glustershd process during spawn.
+func (b *Glustershd) Args() string {
+	shost, _, _ := net.SplitHostPort(config.GetString("clientaddress"))
+	if shost == "" {
+		shost = "localhost"
+	}
+	volFileID := "gluster/glustershd"
+
+	logFile := path.Join(config.GetString("logdir"), "glusterfs", "glustershd.log")
+
+	var buffer bytes.Buffer
+	buffer.WriteString(fmt.Sprintf(" -s %s", shost))
+	buffer.WriteString(fmt.Sprintf(" --volfile-id %s", volFileID))
+	buffer.WriteString(fmt.Sprintf(" -p %s", b.PidFile()))
+	buffer.WriteString(fmt.Sprintf(" -l %s", logFile))
+	buffer.WriteString(fmt.Sprintf(" -S %s", b.SocketFile()))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *replicate*.node-uuid=%s", gdctx.MyUUID))
+
+	b.args = buffer.String()
+	return b.args
+}
+
+// SocketFile returns path to the socket file used for IPC.
+func (b *Glustershd) SocketFile() string {
+
+	if b.socketfilepath != "" {
+		return b.socketfilepath
+	}
+
+	glusterdSockDir := path.Join(config.GetString("rundir"), "gluster")
+	b.socketfilepath = fmt.Sprintf("%s/%x.socket", glusterdSockDir, xxhash.Sum64String(gdctx.MyUUID.String()))
```

Comment:
Oh wait! Isn't this the socket file for glusterd2 itself ?

`SocketFile()` is meant to return the path to socket file that the daemon (glustershd) is listening on. If selh heal daemon doesn't listen on a socket file, return empty string.